scripts/tests/https_server.py

```diff
@@ -0,0 +1,35 @@
+import http.server
+import ssl
+import sys
+import os
+
+def start_server(directory_to_serve, cert_file, key_file, port):
+    if not os.path.exists(directory_to_serve):
+        print(f"Error: The directory {directory_to_serve} does not exist.")
+        return
+
+    current_dir = os.getcwd()
+    cert_file_path = os.path.join(current_dir, cert_file)
+    key_file_path = os.path.join(current_dir, key_file)
+
+    # Change to the specified directory to serve
+    os.chdir(directory_to_serve)
+
+    # Create an SSL context
+    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    context.load_cert_chain(certfile=cert_file_path, keyfile=key_file_path)
+
+    # Configure and start the HTTPS server
+    httpd = http.server.HTTPServer(('localhost', port), http.server.SimpleHTTPRequestHandler)
+    httpd.socket = context.wrap_socket(httpd.socket, server_side=True)
+
+    print(f"Serving on https://localhost:{port} from {directory_to_serve}")
+    while True:
```

Comment:
I didn't expect that python http module doesn't support https, basically what I assume is just using one line of code to start the server.
```
$ python3 -m http.server 8080
```
But since we need to write a python script for configure certificate, I actually would write with Golang, so that we only use one language and lower future maintenance efforts.
Here is a reference article for golang, the main function is quite simple:
https://eli.thegreenplace.net/2021/go-https-servers-with-tls/ 

Btw, this `while True` is suspicion, it has `server.serve_forever()` based on this article. https://realpython.com/python-http-server/#encrypt-the-connection-with-https

I'm sorry it's kind of waste efforts here, but if it's indeed requires introducing code from a different language, it's better bring it into discussion first, of course, you can still bring this into team discuss if insist to keep this python script.

Reply:
Yeah I actually agree with using Golang. I was thinking about it, but since it was easy to do the python server, I just went for it, so will try to do it in Golang and see how it goes